app/scripts/controllers/permissions/index.js

```diff
@@ -323,6 +377,8 @@ export class PermissionsController {
 
   /**
    * Removes the given permissions for the given domain.
+   * Should only be called after confirming that the permissions exist.
```

Comment:
Why is this the case?

Reply:
To avoid sending unnecessary `accountsChanged` notifications. (`rpc-cap` will just no-op for non-existing permissions.)

There's handling for notifications on the inpage provider, so if this method is "misused" it's not a big deal, which is why I'm not bother to check if the permissions exist in this method. The next step would be to do that before sending notifications and calling `removePermissions`.